### ISSUE TITLE
Performance figures?

### ISSUE THREAD

**Issue by EricLBuehler** [RustPython/RustPython#5106]
Hello everyone! I was looking at RustPython, but my cursory glance did not reveal any performance figures. Where can I find such figures?

**Comment by youknowone** [comment]
Hello! We didn't publish any benchmark results. RustPython was 4x-20x slower than CPython in a few benchmarks.
This is one of the latest benchmark: https://github.com/RustPython/RustPython/pull/4615#issuecomment-1465752928

**Comment by EricLBuehler** [comment]
Ok, thanks! I created a Python-like language, [Merlin](https://github.com/EricLBuehler/merlin_rust/tree/master), and was able to achieve some decent performance (about 2x for a simple test case), but slower for more complicated test cases (not shown in the readme). 

Are there any ideas on why this performance deficit occurs? My (qualitative) experience optimizing Merlin indicated that the bounds checks Rust uses on each arithmetic operation, as well as other checks, causes the performance issue. I improved performance by using a register-based VM and `unwrap_unchecked` for options.

Have there been any optimization efforts for `RustPython`? This seems like a really great project, and perhaps it can benefit from Rust's concurrency to become faster than CPython!
